pkg/controller/external_secrets/networkpolicy.go

```diff
@@ -0,0 +1,217 @@
+	if esc.Spec.NetworkPolicies == nil || len(esc.Spec.NetworkPolicies) == 0 {
+		r.log.V(4).Info("No custom network policies configured in ExternalSecretsConfig")
+		return nil
+	}
+
+	for _, npConfig := range esc.Spec.NetworkPolicies {
+		if err := r.createOrApplyCustomNetworkPolicy(esc, npConfig, resourceLabels, externalSecretsConfigCreateRecon); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createOrApplyCustomNetworkPolicy creates or updates a custom network policy based on API configuration.
+func (r *Reconciler) createOrApplyCustomNetworkPolicy(esc *operatorv1alpha1.ExternalSecretsConfig, npConfig operatorv1alpha1.NetworkPolicy, resourceLabels map[string]string, externalSecretsConfigCreateRecon bool) error {
+	// Build the NetworkPolicy object from the API spec
+	networkPolicy := r.buildNetworkPolicyFromConfig(esc, npConfig, resourceLabels)
+
+	networkPolicyName := fmt.Sprintf("%s/%s", networkPolicy.GetNamespace(), networkPolicy.GetName())
+	r.log.V(4).Info("Reconciling custom network policy", "name", networkPolicyName, "component", npConfig.ComponentName)
+
+	fetched := &networkingv1.NetworkPolicy{}
+	exists, err := r.Exists(r.ctx, client.ObjectKeyFromObject(networkPolicy), fetched)
+	if err != nil {
+		return common.FromClientError(err, "failed to check existence of network policy %s", networkPolicyName)
+	}
+
+	if exists {
+		if externalSecretsConfigCreateRecon {
+			r.eventRecorder.Eventf(esc, corev1.EventTypeWarning, "ResourceAlreadyExists", "NetworkPolicy %s already exists", networkPolicyName)
+		}
+		if common.HasObjectChanged(networkPolicy, fetched) {
+			r.log.V(1).Info("NetworkPolicy modified, updating", "name", networkPolicyName)
+			if err := r.UpdateWithRetry(r.ctx, networkPolicy); err != nil {
+				return common.FromClientError(err, "failed to update network policy %s", networkPolicyName)
+			}
+			r.eventRecorder.Eventf(esc, corev1.EventTypeNormal, "Reconciled", "NetworkPolicy %s updated", networkPolicyName)
+		} else {
+			r.log.V(4).Info("NetworkPolicy already up-to-date", "name", networkPolicyName)
+		}
+	} else {
+		if err := r.Create(r.ctx, networkPolicy); err != nil {
+			return common.FromClientError(err, "failed to create network policy %s", networkPolicyName)
+		}
+		r.eventRecorder.Eventf(esc, corev1.EventTypeNormal, "Reconciled", "NetworkPolicy %s created", networkPolicyName)
+	}
+
+	return nil
+}
+
+// createOrApplyNetworkPolicyFromAsset decodes a NetworkPolicy YAML asset and ensures it exists in the cluster.
+func (r *Reconciler) createOrApplyNetworkPolicyFromAsset(esc *operatorv1alpha1.ExternalSecretsConfig, assetName string, resourceLabels map[string]string, externalSecretsConfigCreateRecon bool) error {
+	networkPolicy := common.DecodeNetworkPolicyObjBytes(assets.MustAsset(assetName))
+	updateNamespace(networkPolicy, esc)
+	common.UpdateResourceLabels(networkPolicy, resourceLabels)
```

Comment:
Lets wrap this in a method like `getNetworkPolicyObject`, similar to other resources.